tests/http/httpClient/Extensions/HttpRequestMessageExtensionsTests.cs

```diff
@@ -34,7 +34,7 @@ public async Task GetRequestOptionCanExtractRequestOptionFromHttpRequestMessage(
             };
             requestInfo.AddRequestOptions(new IRequestOption[] { redirectHandlerOption });
             // Act and get a request message
-            var requestMessage = await requestAdapter.ConvertToNativeRequestAsync<HttpRequestMessage>(requestInfo);
+            var requestMessage = (await requestAdapter.ConvertToNativeRequestAsync<HttpRequestMessage>(requestInfo))!;
```

Comment:
could we instead assert not null on the request message? It'll probably make the error message more intelligible if the behaviour changes.

Reply:
Done @baywet - check please [743e8b8](https://github.com/microsoft/kiota-dotnet/pull/329/commits/743e8b8aeeb54de5ff6e43d2b2d2fc94178366da)